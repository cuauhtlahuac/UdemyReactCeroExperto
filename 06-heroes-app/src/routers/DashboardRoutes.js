import React, { Fragment } from 'react';
import { Switch, Route } from 'react-router-dom';

import Navbar from 'components/ui/NavBar';
import DCScreen from 'components/dc/DCScreen';
import MarvelScreen from 'components/marvel/MarvelScreen';
import HeroScreen from 'components/heroes/HeroScreen';

const DashboardRoutes = () => {
	return (
		<Fragment>
			<Navbar />
			
			<div className="container mt-sm-3 mt-xl-5">
				<Switch>
					<Route path="/dc" component={ DCScreen } exact/>
					<Route path="/marvel" component={ MarvelScreen } exact/>
					<Route path="/hero/:heroId" component={ HeroScreen } exact/>
					<Route path="/" exact/>
				</Switch>
			</div>
			 
		</Fragment>
	);
};

export default DashboardRoutes;
