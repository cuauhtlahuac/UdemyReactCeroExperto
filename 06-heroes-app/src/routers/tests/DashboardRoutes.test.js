import React from 'react';
import { mount, shallow } from 'enzyme';
import DashboardRoutes from 'routers/DashboardRoutes';
import { AuthContext } from 'contexts/authContext';
import { MemoryRouter } from 'react-router-dom';

describe('<DashboardRoutes /> tests', () => {
	let component;
	const contextValue = {
		dispatch: jest.fn(),
		user: { logged: true, name: 'Chico Che' },
	};

	beforeEach(() => {
		component = mount(
			<AuthContext.Provider value={contextValue}>
				<MemoryRouter>
					<DashboardRoutes />
				</MemoryRouter>
			</AuthContext.Provider>,
		);
	});

	afterEach(() => {
		component.unmount;
	});

	test('should render correctly', () => {
		expect(component).toMatchSnapshot();
  });
  
  test('should have the user name', () => {
    expect(component.find('span').text()).toBe(contextValue.user.name)
  })
});
