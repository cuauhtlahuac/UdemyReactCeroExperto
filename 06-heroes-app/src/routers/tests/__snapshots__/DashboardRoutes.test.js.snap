// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardRoutes /> tests should render correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark p-3"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Chico Che
        </span>
        <button
          className="nav-item nav-link btn btn-outline-primary"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-sm-3 mt-xl-5"
  >
    <div>
      <h1>
        Marvel
      </h1>
      <div
        className="animate__animated animate__fadeIn row row-cols-1 row-cols-sm-2  row-cols-lg-3 row-cols-xl-4  g-4"
      >
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Spider Man"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Captain America"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Iron Man"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Thor"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Hulk"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Wolverine"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Daredevil"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Hawkeye"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Cyclops"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card mb-1"
            style={
              Object {
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-6"
              >
                <img
                  alt="Silver Surfer"
                  className="animate__animated animate__flipInY card-img"
                  src="../assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-title"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    more...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
